pkg/apis/resource/validation/validation.go

```diff
@@ -875,9 +891,145 @@ func validateDeviceAttributeVersionValue(value *string, fldPath *field.Path) fie
 	return allErrs
 }
 
-func validateDeviceCapacity(capacity resource.DeviceCapacity, fldPath *field.Path) field.ErrorList {
-	// Any parsed quantity is valid.
-	return nil
+// validateMultiAllocatableDeviceCapacity must check requestPolicy in consumable capacity.
+func validateMultiAllocatableDeviceCapacity(capacity resource.DeviceCapacity, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if capacity.RequestPolicy != nil {
+		allErrs = append(allErrs,
+			validateRequestPolicy(capacity.Value, capacity.RequestPolicy, fldPath.Child("requestPolicy"))...)
+	}
+	return allErrs
+}
+
+// validateSingleAllocatableDeviceCapacity must not allow consumable capacity.
+func validateSingleAllocatableDeviceCapacity(capacity resource.DeviceCapacity, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if capacity.RequestPolicy != nil {
+		allErrs = append(allErrs,
+			field.Forbidden(fldPath.Child("requestPolicy"), "allowMultipleAllocations must be true"))
+	}
+	return allErrs
+}
+
+// validateRequestPolicy validates at most one of ValidRequestValues can be defined.
+// If any ValidRequestValues are defined, Default must also be defined and valid.
+func validateRequestPolicy(maxCapacity apiresource.Quantity, policy *resource.CapacityRequestPolicy, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	count := 0
+	if policy.ZeroConsumption != nil {
+		count += 1
+	}
+	if len(policy.ValidValues) > 0 {
+		count += 1
+	}
+	if policy.ValidRange != nil && policy.ValidRange.Min != nil {
+		count += 1
+	}
+	if count > 1 {
+		allErrs = append(allErrs, field.Forbidden(fldPath, `exactly one policy can be specified, cannot specify any of "zeroConsumption", "validValues" and "validRange" at the same time`))
+	} else {
+		allErrs = append(allErrs, validateValidRequestValues(maxCapacity, policy, fldPath)...)
+	}
+	return allErrs
+}
+
+func validateValidRequestValues(maxCapacity apiresource.Quantity, policy *resource.CapacityRequestPolicy, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	switch {
+	case policy.ZeroConsumption != nil && *policy.ZeroConsumption:
+		if policy.Default != nil {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("default"), "default must not be defined when zeroConsumption=true"))
+		}
+	case len(policy.ValidValues) > 0:
+		if policy.Default == nil {
+			allErrs = append(allErrs, field.Required(fldPath.Child("default"), "required when validValues is specified"))
+		} else {
+			allErrs = append(allErrs, validateRequestPolicyValidValues(*policy.Default, maxCapacity, policy.ValidValues, fldPath.Child("validValues"))...)
+		}
+	case policy.ValidRange != nil && policy.ValidRange.Min != nil:
```

Comment:
don't allow setting `validRange:{}` and not providing `min`:

```suggestion
	case policy.ValidRange != nil:
```